### PR TITLE
Concept for cleanup compilation files.

### PR DESCRIPTION
**Pull request by KochynVolodymyr**
> ### Motivate of the pull request
> - [ ] To address an existing issue. If so, please provide a link to the issue: <issue id>
> - [ ] Breaking new feature. If so, please describe details in the description part.

> ### Describe the technical details
> #### What is currently done? (Provide issue link if applicable)
> 
> 
> 
> 
> 
>
> #### What does this pull request change?
> 
> 
> 
> 
> 

> ### Which part of the code base require a change
> 
> - [ ] Library: <Specify the library name>
> - [ ] Plug-in: <Specify the plugin name>
> - [ ] Engine
> - [ ] Documentation
> - [ ] Regression tests
> - [ ] Continous Integration (CI) scripts

> ### Impact of the pull request

> - [ ] Require a change on Quality of Results (QoR)
> - [ ] Break back-compatibility. If so, please list who may be influenced.

### CODE REVIEW

**Comment by KochynVolodymyr** [comment]
@ravic-rs @skyler-rs please review the concept of the cleaning files. But still the main idea is to define a list of files that need to remove.

**Review by skyler-rs — APPROVED** [review]

**Comment by skyler-rs** [comment]
> @ravic-rs @skyler-rs please review the concept of the cleaning files. But still the main idea is to define a list of files that need to remove.

Looks good, that will make updating the clean for a specific task nice and easy. I might rename GetFiles to GetCleanFiles or GetCleanList... something that clarifies what type of files are being returned since this is at the generic Compiler level, but looks good beyond that.

File: src/Compiler/CompilerOpenFPGA.h [hunk 1]
```diff
@@ -125,6 +125,8 @@ class CompilerOpenFPGA : public Compiler {
   virtual bool DesignChanged(const std::string& synth_script,
                              const std::filesystem::path& synth_scrypt_path,
                              const std::filesystem::path& outputFile);
+  virtual std::vector<std::string> GetFiles(
```
Comment (ravic-rs) [inline src/Compiler/CompilerOpenFPGA.h:128]: Options:
GetTaskGenFiles or GetCleanFiles

File: src/Compiler/CompilerOpenFPGA.cpp [hunk 2]
```diff
@@ -789,6 +789,21 @@ bool CompilerOpenFPGA::DesignChanged(
   return result;
 }
 
+std::vector<std::string> CompilerOpenFPGA::GetFiles(
+    Action action, const std::string& projectName) const {
+  std::vector<std::string> files;
+  switch (action) {
+    case Compiler::Action::Analyze:
+      files = {ANALYSIS_LOG, "port_info.json",
```
Comment (ravic-rs) [inline src/Compiler/CompilerOpenFPGA.cpp:797]: Did you look into diff in the folder location for files generated in-between the tasks ?

Comment (ravic-rs) [reply src/Compiler/CompilerOpenFPGA.cpp:797]: I didn't follow. Could you please elaborate with an e.g. ?

**Pull request closed without merging**